packages/core/lib/networks.js

```diff
@@ -1,253 +1,217 @@
-var fs = require("fs");
-var path = require("path");
-var OS = require("os");
-var BlockchainUtils = require("@truffle/blockchain-utils");
-var Provider = require("@truffle/provider");
-var async = require("async");
-var { Web3Shim } = require("@truffle/interface-adapter");
-
-var Networks = {
-  deployed: function(options, callback) {
-    fs.readdir(options.contracts_build_directory, function(err, files) {
-      if (err) {
-        // We can't read the directory. Act like we found nothing.
-        files = [];
-      }
-
-      var promises = [];
-
-      files.forEach(function(file) {
-        promises.push(
-          new Promise(function(accept, reject) {
-            fs.readFile(
-              path.join(options.contracts_build_directory, file),
-              "utf8",
-              function(err, body) {
-                if (err) return reject(err);
-
-                try {
-                  body = JSON.parse(body);
-                } catch (e) {
-                  return reject(e);
-                }
-
-                accept(body);
-              }
-            );
-          })
-        );
-      });
+const fs = require("fs");
+const path = require("path");
+const OS = require("os");
+const BlockchainUtils = require("@truffle/blockchain-utils");
+const Provider = require("@truffle/provider");
+const async = require("async");
+const { Web3Shim } = require("@truffle/interface-adapter");
+
+const Networks = {
+  deployed: async function(options) {
+    let files;
+    try {
+      files = fs.readdirSync(options.contracts_build_directory);
+    } catch (error) {
+      // We can't read the directory. Act like we found nothing.
+      files = [];
+    }
 
-      Promise.all(promises)
-        .then(function(binaries) {
-          var ids_to_names = {};
-          var networks = {};
+    const promises = [];
+
+    files.forEach(file => {
+      promises.push(
+        new Promise((resolve, reject) => {
+          try {
+            const filePath = path.join(options.contracts_build_directory, file);
+            const fileContents = fs.readFileSync(filePath, "utf8");
+            const body = JSON.parse(fileContents);
+            resolve(body);
+          } catch (error) {
+            return reject(error);
+          }
+        })
+      );
+    });
 
-          // binaries.map(function(b) {return b.contract_name + ": " + JSON.stringify(b.networks, null, 2)}).forEach(function(b) {
-          //   console.log(b);
-          // });
+    const binaries = await Promise.all(promises);
```

Comment:
Looks like you can just get rid of a lot of this and do:
```javascript
const binaries = files.map(file => {
  const filePath = path.join(options.contracts_build_directory, file);
  const fileContents = fs.readFileSync(filePath, "utf8");
  const body = JSON.parse(fileContents);
  return body;
});  
```

We don't need the extra layer of Promises because it's not actually doing anything async, and that means we don't need the try/catch because any errors will just end up throwing anyway.